src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -852,6 +852,80 @@ protected theorem extractLsb_ofNat (x n : Nat) (hi lo : Nat) :
     (extractLsb' start len x).getLsbD i = (i < len && x.getLsbD (start+i)) := by
   simp [getLsbD, Nat.lt_succ]
 
+/--
+Get the most significant bit after `extractLsb'`. With `extractLsb'`, we extract
+a `BitVec` `x'` with length `len` from `BitVec` `x`, starting from the element
+at position `start`. The function `getMsb` extracts a bit counting from the most
+significant bit. Assuming certain conditions, `x.(extractLsbD' {w} start len).getMsbD i`
+is equal to `x.getMsbD (w - (start + len - i))`.
+
+Example (w = 10, start := 3, len := 4):
+
+                                |---| = w - (start + len) = 3
+                                      |start + len|       = 7
+                                            |start|       = 3
+                                      | len |             = 4
+let x                       =   9 8 7 6 5 4 3 2 1 0
+let x' = x.extractLsb' 3 4  =         6 5 4 3
+                                      | |
+                                      | x'.getMsbD 1 = x.getMsbD (i := w - (start + len - i) = 10 - (3 + 4 - 1) = 4) = x.getLsbD 5
```

Comment:
```suggestion
                                      | x'.getMsbD 1 =
                                        x.getMsbD (i := w - (start + len - i) = 10 - (3 + 4 - 1) = 4)
```

Should we drop getLsbD, we do not use it at the RHS. Also, maybe let's break the line as the long line wraps on my screen and breaks the figure.